lib/tasks/instrumentation_generator/README.md

```diff
@@ -0,0 +1,42 @@
+# Instrumentation Generator
+
+## Idea
+Create a CLI, similar to Rails’ generators and scaffold scripts, to create the required files and basic classes for instrumenting new libraries. The hope is that the parts of adding new instrumentation that is repetitive can be eliminated and/or reduced so that it takes less time to add new instrumentation and we can spend development time considering the best attributes to collect and interactions to measure. I also hope this project will reduce the overall toil related to adding new instrumentation.
+
+The files we create with little variation besides library and method names are:
+* Lib
+  * Dependency detection
+  * Chain
+  * Prepend
+  * Instrumentation
+* Tests
+  * configuration/newrelic.yml
+  * Envfile
+  * Test file
+
+Furthermore, we also create very similar snippets inside the default source configuration file that could be generated by this project as well.
+
+## Outcome
+A prototype outside the agent has been created that generates the required files to create new instrumentation. This prototype accepts three arguments: name (name of the library), method (method to instrument), args (arguments for the method).
+
+The Ruby gem Thor, a toolkit for building powerful command-line interfaces used in Bundler, Vagrant, Rails and others powers this CLI.
+
+## Usage
+
+`thor list` can be passed to get a list of all available thor commands
+
+### instrumentation:scaffold
```

Comment:
I'm not familiar with thor. Would the command we'd run be `thor instrumentation:scaffold gem_name --method method_name --args first,second`? It might be nice to add an example of exactly what we'd want to run somewhere?

Reply:
Good call. And Kayla's got it covered in `TODO.md` for the future.

Reply:
Great point! I think this seems like a pretty crucial item to add to the README at this stage. I'll incorporate it as part of this review.

Reply:
Resolved in: 5e4deea121cd519bdad166717f6467a65162ff9a